jaeger-core/src/main/java/io/jaegertracing/internal/propagation/BinaryCodec.java

```diff
@@ -0,0 +1,270 @@
+    stream.write((byte) (value >> 8));
+    stream.write((byte) (value));
+  }
+
+  /**
+   * Write an integer into a stream in network order
+   *
+   * @param stream Stream to write the integer into
+   * @param value  integer to write
+   */
+  private static void writeInt(ByteArrayOutputStream stream, int value) {
+    stream.write((byte) (value >> 24));
+    stream.write((byte) (value >> 16));
+    stream.write((byte) (value >> 8));
+    stream.write((byte) (value));
+  }
+
+  /**
+   * Writes a String Key/Value pair into a ByteArrayOutputStream
+   *
+   * @param stream Stream to write the integer into
+   * @param key    key of the KV pair
+   * @param value  value of the KV pair
+   */
+  private void writeKvPair(ByteArrayOutputStream stream, String key, String value) {
+    byte[] buf;
+
+    int keyLen;
+    buf = key.getBytes(DEFAULT_CHARSET);
+    keyLen = buf.length;
+    writeInt(stream, keyLen);
+    stream.write(buf, 0, keyLen);
+
+    int valLen;
+    buf = value.getBytes(DEFAULT_CHARSET);
+    valLen = value.length();
+    writeInt(stream, valLen);
+    stream.write(buf, 0, valLen);
+  }
+
+  /**
+   * Convenience method to check a buffer for size and reallocate if necessary.
+   *
+   * @param len   the length required
+   * @param bytes the buffer of bytes to be used
+   * @return a byte array of the correct size.
+   */
+  private static byte[] checkBuf(int len, byte[] bytes) {
+    return len <= bytes.length ? bytes : new byte[len];
+  }
+
+  @Override
+  public void inject(JaegerSpanContext spanContext, Binary carrier) {
+
+    // Because we need to know the size of a ByteBuffer a priori, we'll
+    // use a stream to serialize and then copy the stream into the
+    // ByteBuffer of the carrier.  The double allocation isn't ideal, but
+    // these should be small and the GC will return this memory very fast.
+    ByteArrayOutputStream stream = new ByteArrayOutputStream(64);
+
+    // Write the IDs
+    writeLong(stream, spanContext.getTraceIdHigh());
+    writeLong(stream, spanContext.getTraceIdLow());
+    writeLong(stream, spanContext.getSpanId());
+    writeLong(stream, spanContext.getParentId());
+
+    // Write the flags (byte)
+    stream.write(spanContext.getFlags());
+
+    // write the baggage count.
+    writeInt(stream, spanContext.baggageCount());
+
+    // write the kv/pars into the stream
+    for (Map.Entry<String, String> entry : spanContext.baggageItems()) {
+      writeKvPair(stream, entry.getKey(), entry.getValue());
+    }
+
+    // Now we have a stream and a size, and we'll copy it into the byte
+    // buffer.
+    int size = stream.size();
+    ByteBuffer buf = carrier.injectionBuffer(size);
+
+    // Java defaults to big endian (network order), but enforce it just
+    // in case the carrier set the wrong byte order before passing it in.
+    if (buf.order() != ByteOrder.BIG_ENDIAN) {
+      throw new IllegalStateException("Carrier byte order must be big endian.");
```

Comment:
Out of curiosity: did you run into this in practice? This check seems strangely too specific :-)

Reply:
I didn't in this case, although did notice that ByteBuffer supports specifying byte order.  I figured it'd be an extremely difficult issue to debug if someone someday implemented a carrier that set it to little endian, so figured this might save someone some time in the future.  ;)